src/book/03-guides/01-main-codebase/02-maintaining-subsystems.mdx

```diff
@@ -0,0 +1,121 @@
+
+1. Build the code, ensuring `ROLE_visualmesh` is set to `ON` in `./b configure -i`. Go to the [Getting Started page](/guides/main/getting-started) to find out more on how to build the code and install to a robot. Ensure the new configuration file has been installed. Check out the [Build System page](/system/foundations/build-system#install) to find out more about options when installing onto the robot.
+
+2. When your new Visual Mesh is installed onto the robot, connect to the robot using
+
+   ```bash
+   ssh nubots@<address>
+   ```
+
+3. Ensure NUsight is on.
+
+   ```bash
+   nano config/NetworkForwarder.yaml
+   ```
+
+   Turn `vision object` and `compressed images` on. Run NUsight using `yarn prod` and navigate to the NUsight page in your browser. More on NUsight can be found on [the NUsight NUbook page](/system/tools/nusight).
+
+4. Run the visualmesh role
+
+   ```bash
+   ./visualmesh
+   ```
+
+5. Wait for the cameras to load and then watch the Vision tab in NUsight. To determine if the output is correct, consult the [vision page](/system/subsystems/vision) for images of the expected output.
+
+If you would like to see the Visual Mesh output in NUsight, you will need to log the data and run it back in NUsight using DataPlayback, since the data is too large to send over a network. See the [DataLogging and DataPlayback guide](/guides/main/data-recording-playback), considering the following:
+
+- Use the [`visualmesh`](https://github.com/NUbots/NUbots/blob/master/roles/visualmesh.role) role to record the data.
+- Set `message.output.CompressedImage` to `true` and add `message.vision.VisualMesh: true` in both [`DataLogging.yaml](https://github.com/NUbots/NUbots/blob/master/module/support/logging/DataLogging/data/config/DataLogging.yaml) and [`DataPlayback.yaml](https://github.com/NUbots/NUbots/blob/master/module/support/logging/DataPlayback/data/config/DataPlayback.yaml).
+- Use the [`playback`](https://github.com/NUbots/NUbots/blob/master/roles/playback.role) role to playback the data, without changes.
+
+### Tuning Detectors
+
+If the Visual Mesh trained with good results but when testing in the system the network did not appear to work very well, you may need to tune the detectors.
+
+[BallDetector.yaml](https://github.com/NUbots/NUbots/blob/master/module/vision/BallDetector/data/config/BallDetector.yaml) and [GoalDetector.yaml](https://github.com/NUbots/NUbots/blob/master/module/vision/GoalDetector/data/config/GoalDetector.yaml) contain the values for tuning the ball and goal detectors respectively.
+
+1. Build and install the `visualmesh` role to a robot.
+2. SSH onto the robot.
+3. Enable NUsight messages on the robot by running
+
+   ```bash
+   nano config/NetworkForwarder.yaml
+   ```
+
+   and set `message.vision.Balls` and `message.vision.Goals` to `true`. This command uses the nano text editor, see this [cheatsheet](https://www.nano-editor.org/dist/latest/cheatsheet.html) for common commands.
```

Comment:
You use nano before this. If you would like to link to the nano cheatsheet, I would consider putting the link at the first usage, although if you're reading this page just for this subsection, then you would miss it :shrug:

Reply:
Hmmmmmm we probably don't need to keep linking to a nano guide every time we use nano in NUbook. Searching 'nano' in NUbook brings up the Getting Started page as the first hit and it has a YouTube guide 🤷